stl/src/primitives.hpp

```diff
@@ -71,26 +43,23 @@ namespace Concurrency {
             CONDITION_VARIABLE m_condition_variable = CONDITION_VARIABLE_INIT;
         };
 
-        inline void create_stl_critical_section(stl_critical_section_win7* p) {
-            new (p) stl_critical_section_win7;
+        // TRANSITION, only used when constexpr mutex constructor is not enabled
+        inline void create_stl_critical_section(void* p) {
+            new (p) _Stl_critical_section;
         }
 
-        inline void create_stl_condition_variable(stl_condition_variable_win7* p) {
+        inline void create_stl_condition_variable(void* p) {
             new (p) stl_condition_variable_win7;
         }

```

Comment:
Changing these parameter types to be `void*` doesn't appear to be necessary, and slightly weakens type safety. (This is only used in `stl/src` so we don't need to worry about `new (p)` selecting something other than True Placement New, that's why aren't globally qualifying it either.)